client/cli/src/lib.rs

```diff
@@ -315,22 +301,67 @@ pub fn init_logger(
 		"%Y-%m-%d %H:%M:%S%.3f".to_string()
 	});
 
-	let subscriber = FmtSubscriber::builder()
+	let subscriber_builder = FmtSubscriber::builder()
 		.with_env_filter(env_filter)
-		.with_writer(std::io::stderr)
+		.with_writer(std::io::stderr as _)
 		.event_format(logging::EventFormat {
 			timer,
+			enable_color,
 			display_target: !simple,
 			display_level: !simple,
 			display_thread_name: !simple,
-			enable_color,
-		})
-		.finish()
-		.with(logging::NodeNameLayer);
+		});
+	if disable_log_reloading {
+		let subscriber = subscriber_builder
+			.finish()
+			.with(logging::NodeNameLayer);
+		initialize_tracing(subscriber, tracing_receiver, profiling_targets)
+	} else {
+		let subscriber_builder = subscriber_builder.with_filter_reloading();
+		let handle = subscriber_builder.reload_handle();
+		sc_tracing::set_reload_handle(handle);
+		let subscriber = subscriber_builder
+			.finish()
+			.with(logging::NodeNameLayer);
+		initialize_tracing(subscriber, tracing_receiver, profiling_targets)
+	}
+}
+
+// Adds default directives to ensure addLogFilter and resetLogFilter RPCs function correctly
+// Panics if any of the provided directives are invalid when `panic_if_invalid` is true.
+fn add_default_directives(mut env_filter: EnvFilter, directives: &str, panic_if_invalid: bool) -> EnvFilter {
```

Comment:
Please no.

This function is soo overcomplicated, please just return a result. The calling code needs to decide if it wants to panic or not.

Reply:
While I don't know how the error message looks like, we should make sure that the people are aware what is wrong and why we return an error. TLDR, have a nice error message.

Reply:
Should be resolved, see above.